ScoreSaber/UI/Leaderboard/ScoreSaberLeaderboardViewController.cs

```diff
@@ -446,62 +621,214 @@ private async Task StartReplay(ScoreMap score) {
+                    cellClicker.ResetColourAndSize();
+
+                    TextMeshProUGUI _playerNameText = tableCell._playerNameText;
+                    TextMeshProUGUI _scoreText = tableCell._scoreText;
+                    TextMeshProUGUI _rankText = tableCell._rankText;
+
+                    _playerNameText.richText = true;
+
+                    if (!obtainedAnchor) {
+                        normalAnchor = _playerNameText.rectTransform.anchoredPosition;
+                        obtainedAnchor = true;
+                    }
+                    Vector2 newPosition = new Vector2(normalAnchor.x + 3f, normalAnchor.y);
+                    _playerNameText.rectTransform.anchoredPosition = newPosition;
+
+                    tableCell.showSeparator = true;
+                    _tweeningService.FadeText(_playerNameText, true, 0.2f);
+                }
+            }
+
+            _downButton.interactable = cells.Length >= 9;
+        }
+
+
+
+
+        public void Initialize() {
+            _infoButtons = new EntryHolder();
+            _scoreDetailView = new ScoreDetailView();
+            _mapInfoView = new MapInfoView();
+            _scoreDetailView.showProfile += scoreDetailView_showProfile;
+            _scoreDetailView.startReplay += scoreDetailView_startReplay;
+            _playerService.LoginStatusChanged += playerService_LoginStatusChanged;
+            _infoButtons.infoButtonClicked += infoButtons_infoButtonClicked;
+            _uploadDaemon.UploadStatusChanged += uploadDaemon_UploadStatusChanged;
+        }
+
         public void Dispose() {
 
-            _platformLeaderboardViewController.didActivateEvent -= LeaderboardViewActivate;
-            _platformLeaderboardViewController.didDeactivateEvent -= LeaderboardViewDeactivate;
             _playerService.LoginStatusChanged -= playerService_LoginStatusChanged;
             _uploadDaemon.UploadStatusChanged -= uploadDaemon_UploadStatusChanged;
             _infoButtons.infoButtonClicked -= infoButtons_infoButtonClicked;
             _scoreDetailView.startReplay -= scoreDetailView_startReplay;
             _scoreDetailView.showProfile -= scoreDetailView_showProfile;
         }
 
-#if PPV3
-
-        public void UpdatePPv3ButtonState(bool active) {
-            buttonPPv3Replay.gameObject.SetActive(active);
+        public void OnLeaderboardSet(BeatmapKey beatmapKey) {
+            _currentBeatmapKey = beatmapKey;
+            try {
+                BeatmapLevel beatmapLevel = _beatmapLevelsModel.GetBeatmapLevel(beatmapKey.levelId);
+                RefreshLeaderboard(beatmapLevel, beatmapKey, leaderboardTableView, currentScoreScope, loadingLB, Guid.NewGuid().ToString()).RunTask();
+            } catch(Exception ex) { Plugin.Log.Error(ex.Message); }
         }
 
-        private LeaderboardScoreData _currentPPv3ScoreData = null;
-        public void CheckPPv3ReplayExists(LeaderboardScoreData scoreData) {
+        internal static class BeatmapDataCache {
```

Comment:
Isn't this just a SpriteCache? BeatmapDataCache to me implies that it would cache more.

Also, same as all the other times: extra classes should be in their own cs files.